frmts/gtiff/geotiff.cpp

```diff
@@ -1132,6 +1132,84 @@ CPLString GTiffGetCompressValues(bool &bHasLZW, bool &bHasDEFLATE,
     return osCompressValues;
 }
 
+/************************************************************************/
+/*                    OGRGTiffDriverGetSubdatasetInfo()                 */
+/************************************************************************/
+
+struct OGRGTiffDriverSubdatasetInfo : public GDALSubdatasetInfo
+{
+  public:
+    explicit OGRGTiffDriverSubdatasetInfo(const std::string &fileName)
+        : GDALSubdatasetInfo(fileName)
+    {
+    }
+
+    // GDALSubdatasetInfo interface
+  private:
+    void parseFileName() override
```

Comment:
what do you think about the opportunity of using this in the GTiffDataset::OpenDir() method to avoid duplication of parsing ? (this could apply to other drivers as well)

Reply:
I'll have a look, I prefer a method that does not have to open the file though, ideally this API should just manipulate strings without I/O overhead.